Sources/MongoSwift/Operations/Operation.swift

```diff
@@ -11,31 +13,69 @@ internal protocol Operation {
 
 /// A protocol for types that can be used to execute `Operation`s.
 internal protocol OperationExecutor {
-    /// Executes an operation using the provided client and optionally provided session.
+    /// Asynchronously executes an operation using the provided client and optionally provided session.
     func execute<T: Operation>(
         _ operation: T,
         using connection: Connection?,
         client: MongoClient,
         session: ClientSession?
-    ) throws -> T.OperationResult
+    ) -> EventLoopFuture<T.OperationResult>
+    /// Asynchronously executes a block of code.
+    func execute<T>(body: @escaping () throws -> T) -> EventLoopFuture<T>
+    /// Closes the executor.
+    func close() -> EventLoopFuture<Void>
 }
 
 /// Default executor type used by `MongoClient`s.
-internal struct DefaultOperationExecutor: OperationExecutor {
+internal class DefaultOperationExecutor: OperationExecutor {
+    /// A group of event loops to use for running operations in the thread pool.
+    private let eventLoopGroup: EventLoopGroup
+    /// The thread pool to execute operations in.
+    private let threadPool: NIOThreadPool
+
+    internal init(eventLoopGroup: EventLoopGroup) {
+        self.eventLoopGroup = eventLoopGroup
+        self.threadPool = NIOThreadPool(numberOfThreads: 5)
```

Comment:
might want to store this "5" in a constant somewhere for easy for easier discoverability, also maybe a comment on how we arrived at this number

Reply:
good catch, this reminded me that I meant to make it an option. I've done that, and made the default value a static constant on `MongoClient`. 
5 doesn't really have a great justification yet so I've opened [SWIFT-705](https://jira.mongodb.org/browse/SWIFT-705) about doing some basic testing to arrive at a good number and documenting why at that point.